src/python-common/ceph/deployment/service_spec.py

```diff
@@ -2176,6 +2199,18 @@ def validate(self) -> None:
                     f'Cannot add ingress: Invalid health_check_interval specified. '
                     f'Valid units are: {valid_units}')
 
+        # validate SSL parametes
+        if self.monitor_ssl:
+            if not self.ssl:
+                raise SpecValidationError(
+                    'To enable SSL for stats, SSL must also be enabled on the frontend.'
+                )
+            if self.monitor_ssl_cert and bool(self.monitor_ssl_cert) != bool(self.monitor_ssl_key):
+                raise SpecValidationError(
+                    'To enable monitor_ssl, both monitor_ssl_cert and monitor_ssl_key '
+                    'must be provided.'
+                )
```

Comment:
If we go with the above method then this check has to be removed as it will fail